nncf/experimental/onnx/algorithms/quantization/min_max_quantization.py

```diff
@@ -73,20 +77,72 @@ def generate_stat_collector(self, quantizer_config: QuantizerConfig) -> TensorSt
+
+            if node.metatype is not OutputNoopMetatype:
+                output.append(node)
+            return False, output
+
+        visited = {node_id: False for node_id in self.nncf_graph.get_all_node_ids()}
+        partial_traverse_function = partial(traverse_function,
+                                            type_check_fn=lambda x: x in COMPLEX_COMPUTATIONAL_LAYERS_METATYPES,
+                                            visited=visited)
+        start_nodes = self.nncf_graph.get_nodes_by_metatypes([OutputNoopMetatype])
+
+        for start_node in start_nodes:
+            for node in self.nncf_graph.traverse_graph(start_node, partial_traverse_function, traverse_forward=False):
+                self.ignored_scopes.append(node.node_name)
+
+    def _add_ignored_scopes_after_topk_nms(self) -> None:
+
+        def traverse_function(node, output, type_check_fn, visited):
+            if visited[node.node_id]:
+                return True, output
+            visited[node.node_id] = True
+
+            if not type_check_fn(node.metatype):
+                if node.metatype not in DETECTION_OUTPUT_FINAL_TYPES:
+                    output.append(node)
+                return False, output
+
+            return True, output
+
+        visited = {node_id: False for node_id in self.nncf_graph.get_all_node_ids()}
+        partial_traverse_function = partial(traverse_function,
+                                            type_check_fn=lambda x: x in COMPLEX_COMPUTATIONAL_LAYERS_METATYPES + [
+                                                OutputNoopMetatype],
+                                            visited=visited)
+
+        start_nodes = self.nncf_graph.get_nodes_by_metatypes(DETECTION_OUTPUT_FINAL_TYPES)
+
+        for start_node in start_nodes:
+            for node in self.nncf_graph.traverse_graph(start_node, partial_traverse_function):
+                self.ignored_scopes.append(node.node_name)
+
     def _get_quantizer_setup(self, model: onnx.ModelProto):
         self.nncf_graph = NNCFGraphFactory.create(model) if self.nncf_graph is None else self.nncf_graph
         ip_graph = InsertionPointGraph(self.nncf_graph)
         pattern = ONNX_HW_FUSED_PATTERNS.get_full_pattern_graph()
         ip_graph = ip_graph.get_ip_graph_with_merged_hw_optimized_operations(pattern)
 
-        weight_nodes = self.nncf_graph.get_nodes_by_metatypes(QUANTIZATION_LAYER_METATYPES)
+        weight_nodes = self.nncf_graph.get_nodes_by_metatypes(WEIGHT_LAYER_METATYPES)
         quantizable_layer_nodes = [QuantizableWeightedLayerNode(weight_node, [QuantizerConfig()]) for weight_node in
                                    weight_nodes]
 
         hw_config_type = self.target_device
         hw_config_path = ONNXHWConfig.get_path_to_hw_config(hw_config_type)
         hw_config = ONNXHWConfig.from_json(hw_config_path)
 
+        self._add_ignored_scopes_after_last_weight_node()
+        self._add_ignored_scopes_after_topk_nms()
```

Comment:
Why should we ignore nodes AFTER the last weight nodes and AFTER TopK & NMS layers separately? I thought we should ignore layers BETWEEN the last weight nodes and TopK/NMS.

Reply:
I reworked this part. Now this algorithm has only one function, which is called `_add_ignored_scopes_detection_output`.
Please, check it out